### ISSUE TITLE
API page links do not work when running behind a proxy

### ISSUE THREAD

**Issue by dougthor42** [dougthor42/trendlines#162]
When running behind a URL proxy (eg: using `URL_PREFIX` and `DispatcherMiddleware`), the URLs for the API doc pages are incorrect.

## Expected:

`/URL_PREFIX/api`
`/URL_PREFIX/api/redoc`

## Actual

`/api`
`/api/redoc`

**Comment by dougthor42** [comment]
It think the root cause of this is the fact that `flask-rest-api` does some magic where it catches all errors and returns them as JSON rather than actual HTTP things that can be redirected.

See https://github.com/Nobatek/flask-rest-api/issues/51 and https://github.com/Nobatek/flask-rest-api/issues/55.

**Comment by dougthor42** [comment]
Fixed in #165. Probably.

**Comment by dougthor42** [comment]
Turns out it's not fixed.

Root cause is the javascript in `core.js`:

```javascript
function treeChanged(e, data) {
  // If `metric_id` is defined, then we can query data
  if (data.node.original.metric_id !== null) {
    var expected = "/api/v1/data/" + data.node.original.metric_id;
    // grab the plot data from the api
    $.getJSON(expected)
      .done(function(jsonData) {
        makePlot(jsonData);

        // This updates the URL to reflect which plot is shown.
        var history_url = "/plot/" + data.node.original.metric_id;
        window.history.pushState('page2', 'Title', history_url);
      })
```
